### ISSUE TITLE
Error: Save failed with the following error: The table '#__finder_tokens' is full

### ISSUE THREAD

**Issue by N8Solutions** [joomla/joomla-cms#24171]
### Steps to reproduce the issue

Make sure the `Smart Search - Content` plugin is enabled.

Save a long article. My terms of service article as raw code is **71,837 characters long**.

Click to Save the article

### Expected result

The article should save without any issues.

### Actual result

The article saves but gives the following error message:

```
Error

Save failed with the following error: The table '#__finder_tokens' is full
```

### System information (as much as possible)

```
=============
System Information
=============
php: Linux 3.10.0-962.3.2.lve1.5.24.9.el7.x86_64 #1 SMP Wed Feb 13 08:24:50 EST 2019 x86_64
dbserver: mysql
dbversion: 5.5.62-MariaDB-cll-lve
dbcollation: utf8_general_ci
dbconnectioncollation: utf8mb4_general_ci
phpversion: 7.2.15
server: Apache
sapi_name: fpm-fcgi
version: Joomla! 3.9.3 Stable [ Amani ] 12-February-2019 15:00 GMT
platform: Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0

=============
PHP Settings
=============
safe_mode: false
display_errors: false
short_open_tag: true
file_uploads: true
magic_quotes_gpc: false
register_globals: false
output_buffering: true
open_basedir: xxxxxx
session.save_path: xxxxxx
session.auto_start: 0
disable_functions: opcache_get_status
xml: true
zlib: true
zip: true
mbstring: true
iconv: true
max_input_vars: 1000

=============
Configuration File
=============
MetaAuthor: 0
MetaDesc: 
MetaKeys: 
MetaRights: 
MetaTitle: 1
MetaVersion: 0
access: 1
cache_handler: file
cachetime: 1
caching: 2
captcha: recaptcha
cookie_domain: 
cookie_path: 
db: xxxxxx
dbprefix: xxxxxx
dbtype: mysqli
debug: 0
debug_lang: 0
display_offline_message: 1
editor: jce
error_reporting: none
feed_email: none
feed_limit: 10
force_ssl: 0
fromname: xxxxxx
ftp_enable: 0
ftp_host: 
ftp_pass: xxxxxx
ftp_port: 21
ftp_root: 
ftp_user: xxxxxx
gzip: 1
helpurl: https://help.joomla.org/proxy?keyref=Help{major}{minor}:{keyref}&lang={langcode}
host: xxxxxx
lifetime: 10080
list_limit: 20
live_site: 
log_path: xxxxxx
mailer: mail
mailfrom: xxxxxx
memcache_compress: 0
memcache_persist: 1
memcache_server_host: xxxxxx
memcache_server_port: 11211
offline: 0
offline_image: 
offline_message: This site is down for maintenance.<br />Please check back again soon.
offset: Europe/Athens
offset_user: UTC
password: xxxxxx
robots: 
secret: xxxxxx
sef: 1
sef_rewrite: 1
sef_suffix: 0
sendmail: xxxxxx
session_handler: database
sitename: xxxxxx
sitename_pagetitles: 2
smtpauth: 0
smtphost: xxxxxx
smtppass: xxxxxx
smtpport: 25
smtpsecure: none
smtpuser: xxxxxx
tmp_path: xxxxxx
unicodeslugs: 0
user: xxxxxx
mailonline: 1
cache_platformprefix: 0
shared_session: 0
memcached_persist: 1
memcached_compress: 0
memcached_server_host: xxxxxx
memcached_server_port: 11211
redis_persist: 1
redis_server_host: xxxxxx
redis_server_port: 6379
redis_server_auth: 
redis_server_db: 0
proxy_enable: 0
proxy_host: 
proxy_port: 
proxy_user: 
proxy_pass: 
massmailoff: 0
replyto: 
replytoname: 
session_memcache_server_host: xxxxxx
session_memcache_server_port: 11211
session_memcached_server_host: xxxxxx
session_memcached_server_port: 11211
frontediting: 1
asset_id: 1
session_redis_server_host: xxxxxx
session_redis_server_port: 6379
session_redis_server_db: 0
session_redis_persist: 1
session_redis_server_auth: 
debug_lang_const: 1

=============
Folder Permissions
=============
administrator/components:  writable
administrator/language:  writable
administrator/language/overrides:  writable
administrator/language/en-GB:  writable
administrator/manifests/libraries:  writable
administrator/manifests/packages:  writable
administrator/manifests/files:  writable
administrator/modules:  writable
administrator/templates:  writable
components:  writable
images:  writable
images/rocketlauncher:  writable
images/banners:  writable
images/paypal:  writable
images/headers:  writable
images/sampledata:  writable
language:  writable
language/overrides:  writable
language/en-GB:  writable
libraries:  writable
media:  writable
modules:  writable
plugins:  writable
plugins/twofactorauth:  writable
plugins/fields:  writable
plugins/finder:  writable
plugins/editors:  writable
plugins/privacy:  writable
plugins/gantry5:  writable
plugins/captcha:  writable
plugins/authentication:  writable
plugins/actionlog:  writable
plugins/user:  writable
plugins/osmap:  writable
plugins/extension:  writable
plugins/search:  writable
plugins/system:  writable
plugins/installer:  writable
plugins/editors-xtd:  writable
plugins/quickicon:  writable
plugins/sampledata:  writable
plugins/content:  writable
templates:  writable
configuration.php:  NOT writable
cache:  writable
administrator/cache:  writable
log:  writable
tmp:  writable
```

### Additional comments

This problem just started happening after I added a few extra paragraphs. The article will save OK when it is roughly **71,191** raw code characters. Too much more than that and it will throw the error message above.

When searching about this issue I found 2 posts about it on the Joomla! Forum.

This one is from 2016
[https://forum.joomla.org/viewtopic.php?t=932536](https://forum.joomla.org/viewtopic.php?t=932536)

This one is slightly more recent, from 2018
[https://forum.joomla.org/viewtopic.php?t=963750](https://forum.joomla.org/viewtopic.php?t=963750)

The gist from both of those is that this issue has been known for quite some time but not fixed yet.

I have found the suggestions in those forum threads worked, that Disabling the `Smart Search - Content` plugin will stop the error from occurring. As of Joomla! v3.9.3 it is not really necessary, as far as I know, to disable the plugin because the article still saves, it just throws the error message when the plugin is enabled. So disabling the plugin only stops the error message from being shown and will instead show that the article was successfully saved.

However, leaving the `Smart Search - Content` plugin enabled, even though the article does get saved, users will have the impression that it was not saved because of the error message shown, and they will start trying to figure out what they need to do in order to save the article.

**Comment by ghost** [comment]
@Hackwar ?

**Comment by Quy** [comment]
Duplicate #22220 ?

**Comment by ghost** [comment]
@N8Solutions can you please look if #22220 is a Duplicate?

**Comment by N8Solutions** [comment]
@Quy & @franz-wohlkoenig Yes, it appears to be a duplicate. Sorry I did not see it when searching.

I can't speak to the search engine issue they describe

> **The second error message during a website crawl**
> 303 ---- / component / search /? Searchpharese = alt & searschword-volcanic eruption and of course on several names

but in regards to the first error message mentioned it is the same as the one I received

```
Error
Save failed with the following error: The table '#__finder_tokens' is full
```

I followed the advice @chrisdavenport gave in the other ticket.
The variable `max_heap_table_size` in phpMyAdmin for the server had not been modified so it was still at the default of **16M**. Simply increasing it to **17M** was enough to save the article without receiving the error message so I went ahead and bumped it up to **20M**.

It's worth noting though that I run my own web hosting business so I have admin access and can make that kind of change on the server where most users will not have that option and will need to find another solution.

It's also possible that the resolution to my other issue, #22098, which was completed after the last comment on issue #22220, contributed to this being able to be fixed by increasing the `max_heap_table_size` variable in phpMyAdmin.

Maybe @10john can comment as to whether or not he is still experiencing his issues?

For myself, this issue is solved. Thank you for your assistance!
